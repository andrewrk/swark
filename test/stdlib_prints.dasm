set push, str
jsr _prints
set push, str2
jsr _prints
set [0x8ffe], 0

:str
    dat 22, "I love cheeseeeseses.", 10
:str2
    dat 6, "12345", 10

:_prints
    set z, sp
    set a, [z+1]      ; pointer to string object
    set b, a
    add b, [b]        ; b points to end of string - 1
    add a, 1          ; a points to first char of string

    :_prints_loop
        ifg a, b
            set pc, _prints_return
        set [0x8fff], [a]     ; print character
        add a, 1
        set pc, _prints_loop
    :_prints_return
    add sp, 2
    set pc, [z]
