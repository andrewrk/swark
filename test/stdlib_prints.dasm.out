I love cheeseeeseses.
12345
